### PR TITLE
Re-implement test_instance_status() using describe-instance-status

### PR DESCRIPTION
**Pull request by wzamazon**
Currently test_instance_status() is implemented using
"aws ec2 wait instance-status-ok", which can fail due to
"Max Attemps Reached".

This patch re-implements the function by running
"aws ec2 describe-instance-status" in a loop, and
set the loop count to 100.

Signed-off-by: Wei Zhang <wzam@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by shijin-aws — APPROVED** [review]

**Comment by shijin-aws** [comment]
Looks good to me, thanks.

File: common.sh [hunk 1]
```diff
@@ -349,10 +349,38 @@ create_instance()
     done
 }
 
-# Holds testing every 15 seconds for 40 attempts until the instance status check is ok
+test_instance_status_once()
+{
+    status=$(aws ec2 describe-instance-status  --instance-ids $1 --query InstanceStatuses[*].InstanceStatus.Status --output text)
```
Comment (dipti-kothari) [inline common.sh:354]: If you are planning to create a list, we will need extra round brackets
status=($(aws ec2 describe-instance-status --instance-ids $1 --query InstanceStatuses[*].InstanceStatus.Status --output text))

Comment (shijin-aws) [reply common.sh:354]: I thought this function is passed one instance-id now. But I am OK with making it flexible.

Comment (dipti-kothari) [reply common.sh:354]: I saw there is a for loop that iterates over status, but I guess we are expecting the status to be string. What would describe-instance-status return if the instance status is not ok. We are checking for ok in the string which would work for both "ok" and "not ok"

Comment (wzamazon) [reply common.sh:354]: For the first question, my test show that bash can loop items in a string variable:

```
38f9d35c04a4:memstat_kernel_1.12.0_round2 wzam$ test="ok ok"
38f9d35c04a4:memstat_kernel_1.12.0_round2 wzam$ for i in $test; do echo $i; done
ok
ok
```

Comment (wzamazon) [reply common.sh:354]: > We are checking for ok in the string which would work for both "ok" and "not ok"
The output of `describe-instance-status` can also be 'initializing'. If that happens, this function still return "not_ok"

Comment (dipti-kothari) [reply common.sh:354]: Got it, makes sense

**Review by dipti-kothari — APPROVED** [review]

**Pull request merged**